### PR TITLE
Allow setting up WiFi

### PR DESCRIPTION
**Pull request by eugulixes**
# Please make sure all below checkboxes in the Mandatory section have been checked before submitting your PR (also don't forget to pay attention to the Depending on Circumstances section)

## Mandatory

- [x] The commit message is an imperative and starts with a capital letter (for example, `Add something`, `Remove something`, `Fix something`, etc.).
- [x] I made sure I hadn't put a period at the end of the the commit message(s).

## Depending on Circumstances

Some of the items in the section become mandatory if you are a first-time contributor and/or your changes are relevant to the items.

- [x] Adding a new parameter I didn't forget to reflect it in `README.md`.
- [x] Adding a new parameter or modifying an existing one I didn't break the alphabetical order.
- [x] Adding a new utility written in Python I didn't forget to reflect it in `pieman/README.md`.
- [x] Adding a new utility written in Python I didn't forget to increase the version number of the [pieman](https://pypi.org/project/pieman/) package in `pieman/setup.py`, `pieman/pieman/__init__.py` and `essentials.sh`.
- [ ] (for first-time contributors) One of the commits in the pull request adds me to the **Acknowledgements** section in `AUTHORS.md`.

# Checklist

- [x] It possible to specify the `ENABLE_WIRELESS` param only for `OS=raspbian-buster-armhf` and `DEVICE=rpi-3-b`. Otherwise, there must be the following error:
  ```
  ENABLE_WIRELESS is not available for the specified device and operating system. OS=raspbian-buster-armhf and DEVICE=rpi-3-b are supported only.
  ```
- [x] It's not possible to specify `WPA_SSID` without `ENABLE_WIRELESS=true`.
- [x] It's not possible to specify `WPA_PSK` without `ENABLE_WIRELESS` and `WPA_SSID`.
- [x] It's possible to specify only `ENABLE_WIRELESS=true` and configure WiFi via raspi-config later.
- [x] It's possible to specify `ENABLE_WIRELESS=true`, `WPA_SSID`, `WPA_PSK` and connect to the specified **private** AP right after the boot.
- [x] It's possible to specify `ENABLE_WIRELESS=true`, `WPA_SSID` and connect to the specified **public** AP right after the boot.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -371,6 +371,22 @@ Enables a custom DNS server specified via this parameter.
 
 Note, that the parameter conflicts with `ENABLE_GOOGLE_DNS`, `ENABLE_BASIC_YANDEX_DNS`, and `ENABLE_FAMILY_YANDEX_DNS`.
 
+##### ENABLE_WIRELESS=false
+
+Enables built-in WiFi (only for Raspberry Pi 3 so far).
```
Comment (denismosolov) [inline README.md]: Raspberry Pi 3 + Raspbian Buster

File: bootstrap/12-pm.sh [hunk 2]
```diff
@@ -50,6 +50,14 @@ elif is_debian_based; then
     add_package_to_includes inetutils-ping
 fi
 
+if ${ENABLE_WIRELESS}; then
+    if is_alpine; then
+        add_package_to_includes wpa_supplicant
```
Comment (denismosolov) [inline bootstrap/12-pm.sh]: let's test on Alpine

File: bootstrap/12-pm.sh [hunk 3]
```diff
@@ -50,6 +50,14 @@ elif is_debian_based; then
...
+    elif is_debian_based; then
+        add_package_to_includes wpasupplicant
```
Comment (denismosolov) [inline bootstrap/12-pm.sh]: Let's test on Devuan and Debian

Comment (eugulixes) [reply bootstrap/12-pm.sh]: `is_debian_based` means Devuan, Debian, Kali and Ubuntu.

File: bootstrap/15-networking.sh [hunk 4]
```diff
@@ -61,7 +61,24 @@ sed -i "s/{HOSTNAME}/${HOST_NAME}/" "${ETC}/hostname"
...
-install_readonly files/network/interfaces ${ETC}/network/interfaces
+render "${PIEMAN_DIR}"/files/network/interfaces.j2 ${ETC}/network/interfaces
+
+if ${ENABLE_WIRELESS}; then
+    if is_debian_based; then
+        echo brcmfmac > "${ETC}"/modules-load.d/networking.conf
+
+        mkdir -p "${R}"/lib/firmware/brcm
+
+        wget https://raw.githubusercontent.com/RPi-Distro/firmware-nonfree/86e88fbf0345da49555d0ec34c80b4fbae7d0cd3/brcm/brcmfmac43430-sdio.bin -O "${R}"/lib/firmware/brcm/brcmfmac43430-sdio.bin
```
Comment (denismosolov) [inline bootstrap/15-networking.sh]: I'd install firmware-brcm80211 instead

File: pieman.sh [hunk 5]
```diff
@@ -202,6 +208,24 @@ check_mutually_exclusive_params \
     CREATE_ONLY_CHROOT \
     ENABLE_MENDER
 
+if [[ -n ${WPA_PSK} ]]; then
+    check_if_wpa_psk_is_valid
+fi
```
Comment (denismosolov) [inline pieman.sh]: I'd move below dependencies check

File: bootstrap/15-networking.sh [hunk 6]
```diff
@@ -61,7 +61,24 @@ sed -i "s/{HOSTNAME}/${HOST_NAME}/" "${ETC}/hostname"
 install_readonly files/etc/hosts.template ${ETC}/hosts
 sed -i "s/{HOSTNAME}/${HOST_NAME}/" "${ETC}/hosts"
 
-install_readonly files/network/interfaces ${ETC}/network/interfaces
+render "${PIEMAN_DIR}"/files/network/interfaces.j2 ${ETC}/network/interfaces
+
+if ${ENABLE_WIRELESS}; then
+    if is_debian_based; then
+        echo brcmfmac > "${ETC}"/modules-load.d/networking.conf
```
Comment (denismosolov) [inline bootstrap/15-networking.sh]: I'd rename networking.conf to wireless.conf

**Pull request merged**